packages/tools/src/tools/base/ContourSegmentationBaseTool.ts

```diff
@@ -0,0 +1,355 @@
+        enabledElement,
+        targetId,
+        annotation,
+        annotationStyle,
+        svgDrawingHelper,
+      });
+
+      renderStatus ||= annotationRendered;
+    }
+
+    return renderStatus;
+  }
+
+  /**
+   * Check if an annotation is a contour segmentation annotation.
+   *
+   * This is helpful on classes that can work in a hybrid way as a normal
+   * annotation tool or as a segmentation tool such as spline ROI tool. In that
+   * the segmentation must not be registered/unregistered as a contour
+   * segmentation annotation.
+   *
+   * @param annotation - Annotation
+   * @returns True if the annotation is a contour segmentation annotation or false otherwise
+   */
+  protected isSegmentationAnnotation(
+    annotation: Annotation
+  ): annotation is SegmentationAnnotation {
+    return (
+      (annotation as SegmentationAnnotation).segmentationData !== undefined
+    );
+  }
+
+  /**
+   * Return the segmentation data that needs to be merged with annotation data
+   * in any child class to compose the annotation data.
+   * @returns Annotation segmentation data
+   */
+  protected getSegmentationAnnotationData(): SegmentationAnnotationData {
+    const { toolGroupId } = this;
+    const activeSegmentationRepresentation =
+      activeSegmentation.getActiveSegmentationRepresentation(toolGroupId);
+
+    if (!activeSegmentationRepresentation) {
+      throw new Error(
+        'No active segmentation detected, create one before using scissors tool'
+      );
+    }
+
+    const { type: segmentationType } = activeSegmentationRepresentation;
+
+    if (segmentationType !== SegmentationRepresentations.Contour) {
+      throw new Error(`A contour segmentation must be active`);
+    }
+
+    const { segmentationId, segmentationRepresentationUID } =
+      activeSegmentationRepresentation;
+    const segmentIndex =
+      segmentIndexController.getActiveSegmentIndex(segmentationId);
+
+    return {
+      segmentationData: {
+        segmentationId,
+        segmentIndex,
+        segmentationRepresentationUID,
+      },
+    };
+  }
+
+  protected addAnnotation(annotation: Annotation, element: HTMLDivElement) {
+    const isSegmentationAnnotation = this.isSegmentationAnnotation(annotation);
+
+    // annotationUID is available only after calling addAnnotation
+    addAnnotation(annotation, element);
+
+    if (isSegmentationAnnotation) {
+      this._registerContourSegmentationAnnotation(annotation);
+    }
+  }
+
+  /**
+   * Unregister the segmentation when the annotation is canceled
+   * @param annotation - Contour segmentation annotation
+   */
+  protected cancelAnnotation(annotation: Annotation): void {
+    const isSegmentationAnnotation = this.isSegmentationAnnotation(annotation);
+
+    if (isSegmentationAnnotation) {
+      this._unregisterContourSegmentationAnnotation(annotation);
+    }
+  }
+
+  /**
+   * Return the annotation style based on global, toolGroup, segmentation
```

Comment:
Could you add a line here that explains why you are returning it as an object - I think something like:
Return the annotation style .... **so that it can be used for passing the style into ..... for ...**
I was thinking you needed it for vtk, but I'm not so sure now.

Reply:
Comment updated.

Reply:
Comment updated